### ISSUE TITLE
Log, but not notify

### ISSUE THREAD

**Issue by dok** [bugsnag/bugsnag-js#492]
Hello, we currently utilize this setup:

```node
const bugsnag = require('@bugsnag/js');
const bugsnagExpress = require('@bugsnag/plugin-express');
const config = require('config');

const env = process.env.NODE_ENV || 'development';

module.exports = (releaseStage = env) => {
  const bugsnagClient = bugsnag({
    apiKey: config.bugsnagKey,
    releaseStage,
    notifyReleaseStages: ['production', 'staging', 'pr'],
    appVersion: config.herokuReleaseVersion,
    logger: env !== 'test' ? undefined : null, // Use default, but supress logs in test
  });
  bugsnagClient.use(bugsnagExpress);
  return bugsnagClient;
};
```

In development, we would like any errors to propagate. Instead, we are seeing:
![Screen Shot 2019-03-11 at 4 59 37 PM](https://user-images.githubusercontent.com/1214868/54165698-147e3080-441f-11e9-9660-75ba736e15e6.png)

We don't notify when the `releaseStage` is development, because we have channels that automatically notify other channels that listen for bugsnag reports. Essentially, we only log `production` and `staging` errors.

Is this possible? I'm guessing the `logger` option is a safe bet on doing something like this, but I don't see any documentation on how to use the functions specifically.

Any help would be great. Thank you!

**Comment by bengourley** [comment]
You can use [`autoNotify: false`](https://docs.bugsnag.com/platforms/javascript/configuration-options/#autonotify) to prevent Bugsnag from attaching the event handlers which capture unhandled exceptions/rejections, allowing the errors to propagate in the default way.

However, any manual calls to `bugsnagClient.notify(err)` would still show the log message you're seeing, and not the error. To get logs for those, the simplest thing to do it wrap `bugsnag.notify()`:

```js
function notify (err, opts) {
  if (env === 'development') console.error(err)
  bugsnagClient.notify(err, opts)
}
```

Hope this helps. Any other questions like this, it's best to write in to support@bugsnag.com as GitHub issues are used mainly for bugs with the notifier. Thanks!

**Comment by gaastonsr** [comment]
Is that the recommended approach, overriding the `notify` function? I wish there was a different way.

By the way, sending questions to support@bugsnag.com will reduce the visibility of this type of questions that can easily help other newcomers.

**Comment by mattdyoung** [comment]
Hi @gaastonsr 

I've just tried this with the latest bugsnag-js and our [example express app](https://github.com/bugsnag/bugsnag-js/tree/master/examples/express) and the library outputs the error to the console as well as not sending it to Bugsnag:
```
[bugsnag] Loaded!
Example app listening at http://localhost:9871
[bugsnag] Report not sent due to releaseStage/notifyReleaseStages configuration
Error: Invalid DSL syntax
    at /Users/matt/dev/bugsnag-js/examples/express/app.js:29:9
    at Layer.handle [as handle_request] (/Users/matt/dev/bugsnag-js/examples/express/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/matt/dev/bugsnag-js/examples/express/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/Users/matt/dev/bugsnag-js/examples/express/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/Users/matt/dev/bugsnag-js/examples/express/node_modules/express/lib/router/layer.js:95:5)
    at /Users/matt/dev/bugsnag-js/examples/express/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/Users/matt/dev/bugsnag-js/examples/express/node_modules/express/lib/router/index.js:335:12)
    at Domain.next (/Users/matt/dev/bugsnag-js/examples/express/node_modules/express/lib/router/index.js:275:10)
    at Domain.run (domain.js:359:14)
    at requestHandler (/Users/matt/dev/bugsnag-js/examples/express/node_modules/@bugsnag/plugin-express/dist/bugsnag-express.js:787:18)
```

Can you confirm whether you're still seeing errors not propagating and being logged to the console with the latest version?

We improved the behaviour in https://github.com/bugsnag/bugsnag-js/pull/515 released in v6.1.0.
